node/src/effect.rs

```diff
@@ -1183,48 +1184,38 @@ impl<REv> EffectBuilder<REv> {
             .await
     }
 
-    /// Checks whether the deploys included in the block exist on the network. This includes
-    /// the block's timestamp, in order that it be checked against the timestamp of the deploys
-    /// within the block.
-    pub(crate) async fn validate_block<I>(
-        self,
-        sender: I,
-        block: Block,
-        block_timestamp: Timestamp,
-    ) -> (bool, Block)
+    /// Checks whether the deploys included in the block exist on the network and the block is
+    /// valid.
+    pub(crate) async fn validate_block<I>(self, sender: I, block: Block) -> bool
```

Comment:
Chatting with @afck , let's have this be generic in `Into<ValidatingBlock>`.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/1314/commits/a1f34f21f5632b03aeb0da8b85588ba348409c47.